### PR TITLE
Vector cleanups

### PR DESCRIPTION
**Pull request by Denneisk**
- Added `lerp(vector, vector, number)`
- Optimized `min` and `max` to use square magnitude instead
- Optimized `vector:toAngle`
- Changed `randVec` to use `Vector.Random`
- Changed parameter names on `mix` to be more descriptive
- Changed `mix` to use `LerpVector`
- Changed `ANGLE_ZERO` constant to use preexisting `angle_zero` constant
- Remove temporary bezier fix

Fixes #2893

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/vector.lua [hunk 1]
```diff
@@ -432,16 +414,12 @@ end
...
+e2function vector min(vector vec1, vector vec2)
+	return vec1:LengthSqr() < vec2:LengthSqr() and vec1 or vec2
 end
 
-e2function vector max(vector rv1, vector rv2)
-	local length1 = ( rv1[1] * rv1[1] + rv1[2] * rv1[2] + rv1[3] * rv1[3] ) ^ 0.5
-	local length2 = ( rv2[1] * rv2[1] + rv2[2] * rv2[2] + rv2[3] * rv2[3] ) ^ 0.5
-	if length1 > length2 then return rv1 else return rv2 end
+e2function vector max(vector vec1, vector vec2)
+return vec1:LengthSqr() > vec2:LengthSqr() and vec1 or vec2
```
Comment (Vurv78) [inline lua/entities/gmod_wire_expression2/core/vector.lua:422]: Missing indentation

**Review by Vurv78 — CHANGES_REQUESTED** [review]
lgtm, was gonna say maybe lerp(nvv) to match gmod's function but looks like unity, godot and unreal do lerp(from, to, frac) so better to imitate that.

**Review by Vurv78 — APPROVED** [review]

File: lua/entities/gmod_wire_expression2/core/vector.lua [hunk 2]
```diff
@@ -432,16 +414,12 @@ end
...
+e2function vector min(vector vec1, vector vec2)
+	return vec1:LengthSqr() < vec2:LengthSqr() and vec1 or vec2
 end
 
-e2function vector max(vector rv1, vector rv2)
-	local length1 = ( rv1[1] * rv1[1] + rv1[2] * rv1[2] + rv1[3] * rv1[3] ) ^ 0.5
-	local length2 = ( rv2[1] * rv2[1] + rv2[2] * rv2[2] + rv2[3] * rv2[3] ) ^ 0.5
-	if length1 > length2 then return rv1 else return rv2 end
+e2function vector max(vector vec1, vector vec2)
+	return vec1:LengthSqr() > vec2:LengthSqr() and vec1 or vec2
```
Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/core/vector.lua:422]: These functions are really stupid. Why don't they max/min the components? Is there a different function that does that?

Comment (Denneisk) [reply lua/entities/gmod_wire_expression2/core/vector.lua:422]: Yes, `maxVec` declared right below this

Comment (Vurv78) [reply lua/entities/gmod_wire_expression2/core/vector.lua:422]: Why would you even need that?

Comment (Denneisk) [reply lua/entities/gmod_wire_expression2/core/vector.lua:422]: I could see uses for both.

**Comment by Vurv78** [comment]
> congratulation, you successfully broke the bezier function.

Quit bitching, your server is 3 months outdated

**Pull request merged**